relay-cabi/src/processing.rs

```diff
@@ -393,6 +393,37 @@ pub unsafe extern "C" fn relay_compare_versions(a: *const RelayStr, b: *const Re
     }
 }
 
+fn cmp_no_build_code(
+    ver_a: &sentry_release_parser::Version,
+    ver_b: &sentry_release_parser::Version,
+) -> Ordering {
+    match ver_a.quad().cmp(&ver_b.quad()) {
+        Ordering::Equal => match (ver_a.pre(), ver_b.pre()) {
+            (None, Some(_)) => Ordering::Greater,
+            (Some(_), None) => Ordering::Less,
+            (Some(self_pre), Some(other_pre)) => self_pre.cmp(other_pre),
+            (None, None) => Ordering::Equal,
+        },
+        other => other,
+    }
+}
+
+/// Compares two versions, ignoring build codes.
+#[unsafe(no_mangle)]
+#[relay_ffi::catch_unwind]
+pub unsafe extern "C" fn relay_compare_versions_no_build_code(
+    a: *const RelayStr,
+    b: *const RelayStr,
+) -> i32 {
+    let ver_a = sentry_release_parser::Version::parse(unsafe { (*a).as_str() })?;
+    let ver_b = sentry_release_parser::Version::parse(unsafe { (*b).as_str() })?;
+    match cmp_no_build_code(&ver_a, &ver_b) {
```

Comment:
We could call [`as_semver`](https://docs.rs/sentry-release-parser/latest/sentry_release_parser/struct.Version.html#method.as_semver) on the versions here, and then call `cmp_precedence` on the `semver::Version`, which does what you describe in the PR description:

> Compare the major, minor, patch, and pre-release value of two versions, disregarding build metadata. Versions that differ only in build metadata are considered equal. This comparison is what the SemVer spec refers to as “precedence”.

https://docs.rs/semver/latest/semver/struct.Version.html#method.cmp_precedence

This would require enabling the `semver` feature in the release parser dependency: https://github.com/getsentry/relay/blob/3e01f712801847986249f8f3aedbc68c1d767035/relay-cabi/Cargo.toml#L36